src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerResponseErrorHandler.java

```diff
@@ -63,7 +62,8 @@ private static String concatenateErrorMessages(List<Map<String, Object>> errors)
 
     @Override
     public void handleError(ClientHttpResponse response) throws IOException {
-        HttpStatus statusCode = response.getStatusCode();
+        HttpStatus statusCode = HttpStatus.valueOf(response.getStatusCode()
+                                                           .value());
```

Comment:
why is it needed to use valueOf

Reply:
response.getStatusCode() returns HttpStatusCode in Spring 6, hence why the conversion is needed